.evergreen/config_generator/components/cse/winssl.py

```diff
@@ -0,0 +1,76 @@
+from shrub.v3.evg_build_variant import BuildVariant
+from shrub.v3.evg_task import EvgTaskRef
+
+from config_generator.etc.compile import generate_compile_tasks
+
+from config_generator.etc.cse.compile import CompileCommon
+from config_generator.etc.cse.test import generate_test_tasks
+
+
+SSL = 'winssl'
+TAG = f'cse-matrix-{SSL}'
+
+
+# pylint: disable=line-too-long
+# fmt: off
+COMPILE_MATRIX = [
+    ('windows-64-vs2015', 'vs2015x64', None, ['cyrus']),
+    ('windows-64-vs2017', 'vs2017x64', None, ['cyrus']),
+]
+
+# TODO (CDRIVER-3789): test cse with the 'sharded' topology.
+TEST_MATRIX = [
+    ('windows-64-vs2017', 'vs2017x64', None, 'cyrus', ['auth'], ['server'], ['4.2', '4.4', '5.0']),
+
+    # Test 6.0+ with a replica set since Queryable Encryption does not support the 'server' topology.
+    ('windows-64-vs2017', 'vs2017x64', None, 'cyrus', ['auth'], ['server', 'replica' ], ['6.0', 'latest']),
```

Comment:
The [combination of CSE + 6.0 and latest tests](https://spruce.mongodb.com/version/63e5517461837d4373055658/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&taskName=cse-.%2A-vs2017-x64-test) seems to exceed the current exec_timeout_secs of 1 hour. Should we bump exec_timeout_secs for Windows test tasks up to 2 hours? Or can add `--no-fork` when executing `test-libmongoc` to speed up execution? (Or another workaround?)

Reply:
Will be addressed by restricting executed tests by CSE tasks to `/client_side_encryption/*`.